editor/src/components/canvas/ui-jsx-canvas.spec.tsx

```diff
@@ -2041,6 +2041,34 @@ export var ${BakedInStoryboardVariableName} = (props) => {
     </Storyboard>
   )
 }
+`,
+    )
+  })
+
+  it('React.useEffect at the root fails usefully', () => {
```

Comment:
could you also test the error the canvas throws?

Reply:
So slightly weirdly in this test it doesn't throw an error, so perhaps this description should be rewritten.

Reply:
Now in looking at this it _appears_ that under the hood something is ping-ponging the canvas (between shown and not) when the error is displayed, which makes me think that there's some React shenanigans going on. I'm tempted to push this off into the follow on ticket to get the proper error to display.